types/views/views.go

```diff
@@ -132,6 +132,39 @@ func (v Slice[T]) AsSlice() []T {
 	return v.AppendTo(v.ж[:0:0])
 }
 
+// IndexFunc returns the first index of an element in v satisfying f(e),
+// or -1 if none do.
+func (v Slice[T]) IndexFunc(f func(T) bool) int {
+	for i := 0; i < v.Len(); i++ {
+		if f(v.At(i)) {
+			return i
+		}
+	}
+	return -1
+}
+
+// ContainsFunc reports whether any element in v satisfies f(e).
+func (v Slice[T]) ContainsFunc(f func(T) bool) bool {
+	for i := 0; i < v.Len(); i++ {
+		if f(v.At(i)) {
+			return true
+		}
+	}
+	return false
+}
+
+// SliceContains reports whether v contains element e.
+//
+// As it runs in O(n) time, use with care.
+func SliceContains[T comparable](v Slice[T], e T) bool {
```

Comment:
Put this on the Slice itself?

Reply:
can't. slices aren't necessarily comparable and can't add constraints on a method: https://go.dev/play/p/pNj8OAApOGt